src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.MultipleStatementsCodeGenerator.cs

```diff
@@ -32,23 +28,6 @@ public sealed class MultipleStatementsCodeGenerator(
                 CSharpCodeGenerationOptions options,
                 bool localFunction) : CSharpCodeGenerator(insertionPoint, selectionResult, analyzerResult, options, localFunction)
             {
-                public static bool IsExtractMethodOnMultipleStatements(SelectionResult code)
```

Comment:
this logic moved from the generator to the 'selection result' type (where it makes much more sense) allowing certain checks to be front loaded *prior* to the generation step.